### ISSUE TITLE
Metro crashes often on Linux

### ISSUE THREAD

**Issue by agazso** [status-im/status-react#18211]
# Problem

I am running `make run-metro` on Linux, but it crashes often:

```
/home/attila/Projects/LogosLab/status-mobile/node_modules/nullthrows/nullthrows.js:9
  throw error;
  ^

Error: Got unexpected null
    at nullthrows (/home/attila/Projects/LogosLab/status-mobile/node_modules/nullthrows/nullthrows.js:7:15)
    at Timeout.emitChange [as _onTimeout] (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/index.js:823:48)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  framesToPop: 1
}

Node.js v18.16.0
```

**Comment by agazso** [comment]
Another case:

```
node:internal/fs/watchers:255
    throw error;
    ^

Error: ENOENT: no such file or directory, watch '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm'
    at FSWatcher.<computed> (node:internal/fs/watchers:247:19)
    at Object.watch (node:fs:2343:34)
    at NodeWatcher._watchdir (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/NodeWatcher.js:145:24)
    at /home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/NodeWatcher.js:264:22
    at Walker.<anonymous> (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/common.js:127:31)
    at Walker.emit (node:events:513:28)
    at /home/attila/Projects/LogosLab/status-mobile/node_modules/walker/lib/walker.js:69:16
    at FSReqCallback.oncomplete (node:fs:198:23) {
  errno: -2,
  syscall: 'watch',
  code: 'ENOENT',
  path: '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm',
  filename: '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm'
}

Node.js v18.16.0
make: *** [Makefile:268: run-metro] Error 1

```

**Comment by agazso** [comment]
After restart it asks for manual confirmation of deletion:

```
Copying node_modules from Nix store:
 - /nix/store/v279wdblqar1p8l2w1p8a3g847c21lh5-status-mobile-node-deps-1.25.0-patched
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cmake/debug/obj/x86_64/libc++_shared.so'? y
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cmake/debug/obj/arm64-v8a/libc++_shared.so'? y
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cmake/debug/obj/x86/libc++_shared.so'? y
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cxx/Debug/723c3r5j/obj/x86_64/libc++_shared.so'? y
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cxx/Debug/723c3r5j/obj/arm64-v8a/libc++_shared.so'? y
rm: remove write-protected regular file '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-reanimated/android/build/intermediates/cxx/Debug/723c3r5j/obj/x86/libc++_shared.so'? y
Done in: 27.811s
```

**Comment by siddarthkay** [comment]
There are 2 issues here,

- 1st is the frequent crashes of the metro terminal :
metro crash is related to the version of metro we currently have and its directly coupled with our react-native version i.e `0.72.5`
Issue in metro repository : https://github.com/facebook/metro/issues/1015 is fixed and we should get this fix soon as a patch version to `0.72.x`

This fix is already present in react-native `0.73.0` , however upgrading to that will take a while.

- 2nd is these write protected files, which I will send out a PR for soon.

Thanks for reporting @agazso !

**Comment by siddarthkay** [comment]
issue 1 is also tracked here : https://github.com/status-im/status-mobile/issues/18166
Issue 2 is fixed with https://github.com/status-im/status-mobile/pull/18214
